app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.ts

```diff
@@ -118,11 +147,13 @@ export function validateAddEthereumChainParams(params, end) {
     );
   };
 
-  const firstValidRPCUrl = rpcUrls.find((rpcUrl) => isLocalhostOrHttps(rpcUrl));
+  const firstValidRPCUrl = rpcUrls.find((rpcUrl) =>
+    isLocalhostOrHttps(rpcUrl as string),
+  );
```

Comment:
No need for a type assertion here when we can simply validate.

```suggestion
  const firstValidRPCUrl = rpcUrlsfind((rpcUrl) => 
  	typeof rpcUrl === 'string' && isLocalhostOrHttps(rpcUrl),
  );
```